### PR TITLE
L1TrackTrigger TTTrackmods CMSSW_10

### PR DESCRIPTION
**Pull request by mdhildreth**
#### PR description:

Add digitized TTTrack variables to TTTrack class.  Add new interfaces.  Keep old interfaces for backwards compatibility.

#### PR validation:

Purely technical fixes.  Old interfaces remain.

#### if this PR is a backport please specify the original PR:

Backport of 27686

### CODE REVIEW

**Comment by mdhildreth** [comment]
please test

**Comment by mdhildreth** [comment]
please test

**Comment by fioriNTU** [comment]
+1

**Comment by fabiocos** [comment]
@mdhildreth which production use case is addressing this backport? Generically analysis and studies of the L1T TDR samples?

File: DataFormats/L1TrackTrigger/interface/TTTrack.h [hunk 1]
```diff
@@ -14,479 +14,381 @@
 #define L1_TRACK_TRIGGER_TRACK_FORMAT_H
 
 #include "DataFormats/L1TrackTrigger/interface/TTStub.h"
+#include "DataFormats/L1TrackTrigger/interface/TTTrack_TrackWord.h"
 #include "FWCore/MessageLogger/interface/MessageLogger.h"
 #include "DataFormats/Common/interface/DetSetVectorNew.h"
 #include "DataFormats/TrackerCommon/interface/TrackerTopology.h"
+#include "DataFormats/SiStripDetId/interface/StripSubdetector.h"
```
Comment (davidlange6) [inline DataFormats/L1TrackTrigger/interface/TTTrack.h]: hi @mdhildreth  - what change requires this include?

Comment (mdhildreth) [reply DataFormats/L1TrackTrigger/interface/TTTrack.h]: Hi @davidlange6. Ah, maybe this isn't required in 10X.  It definitely is in 11X, as someone reshuffled the headers.  I could strike it here if it's a concern.

**Comment by mdhildreth** [comment]
@fabiocos These changes are requested by the TrackTrigger folks for L1 TDR production.

**Comment by davidlange6** [comment]
strange - what is the error you get without it (i 11x)? (this header should be used minimally at best now)

File: DataFormats/L1TrackTrigger/interface/TTTrack.h (hunk 1 above)
Comment (mdhildreth) [reply DataFormats/L1TrackTrigger/interface/TTTrack.h]: @davidlange6 In 11X the code didn't compile without that header file (I found this out by copying the 10X version an 11X release area first). I agree it's probably not needed here.

**Comment by kpedro88** [comment]
backport of #27686

**Comment by kpedro88** [comment]
@rekovic are there plans to do further production for L1T TDR with new 10_6_X versions?

**Comment by mdhildreth** [comment]
please test

**Comment by jfernan2** [comment]
+1

**Comment by mdhildreth** [comment]
@kpedro88 @fabiocos @rekovic I updated this backport to match the one previously put in the master branch. (With the same caveats of future cleanup once the L1 TDR is done.)  Could you please sign off on this?

**Comment by kpedro88** [comment]
@mdhildreth is there a clear use case for this backport? Another round of production, user analysis, etc.?

**Comment by mdhildreth** [comment]
@kpedro88 My understanding is that the L1Track stuff is going to be used "post-production" on the 10_6 samples.  So, it's just to have things close in release, rather than using CMSSW_11.

**Comment by kpedro88** [comment]
+upgrade

**Comment by rekovic** [comment]
+1

**Comment by fabiocos** [comment]
+1

**Pull request merged**